google/cloud/bigtable/internal/table.h

```diff
@@ -149,11 +150,9 @@ class Table {
         "The arguments passed to ReadModifyWriteRow(row_key,...) must be "
         "convertible to bigtable::ReadModifyWriteRule");
 
-    // TODO(#336) - optimize this code by not copying the parameter pack.
-    // Add first default rule
     *request.add_rules() = rule.as_proto_move();
     // Add if any additional rule is present
-    std::initializer_list<bigtable::ReadModifyWriteRule> rule_list{
+    std::initializer_list<bigtable::ReadModifyWriteRule>&& rule_list{
```

Comment:
I do not think that works either:

https://en.cppreference.com/w/cpp/utility/initializer_list

only has `const` iterators, there is no way to move the rules out of the list.  The way to avoid copying here is to recurse through the parameter pack:

```C++
Row ReadModifyWriteRow(...) {
...
*request.add_rules() = rule.as_proto_move();
AddRules(request, std::forward<Args>(args)...);
}

private:
  void AddRules(google::bigtable::v2::ReadModifyWriteRowRequest& request) {
    // no-op for empty list
  }
 
  template <typename... Args> 
  void AddRules(google::bigtable::v2::ReadModifyWriteRowRequest& request,
                           bigtable::ReadModifyWriteRule rule, Args&&... args) {
    *request.add_rules() = std::move(rule);
    AddRules(request, std::forward<Args>(args)...);
  }
```